src/arrayMethodSort.js

```diff
@@ -3,9 +3,35 @@
 /**
  * Implement method Sort
  */
+function defaultFunction(a, b) {
+  if (String(a) > String(b)) {
+    return 1;
+  } else {
+    return -1;
+  }
+}
+
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
+  [].__proto__.sort2 = function(compareFunction = defaultFunction) {
     // write code here
+    let isStep;
+
+    do {
+      isStep = 0;
```

Comment:
The variable is named like a boolean but stores a number. Fix it or rename the variable